From 96bb9953f2d3e0acd4416b4c10637555882f2326 Mon Sep 17 00:00:00 2001
From: Martin Schrodt <martin@schrodt.org>
Date: Sun, 15 Oct 2017 01:22:55 +0200
Subject: [PATCH 12/30] try better correction

---
 hw/audio/hda-codec.c | 32 +++++++++++++++++++++-----------
 1 file changed, 21 insertions(+), 11 deletions(-)

diff --git a/hw/audio/hda-codec.c b/hw/audio/hda-codec.c
index 60a5e40827..872840f602 100644
--- a/hw/audio/hda-codec.c
+++ b/hw/audio/hda-codec.c
@@ -182,6 +182,8 @@ struct HDAAudioState {
 
 #define dolog(fmt, ...) AUD_log("", fmt, ## __VA_ARGS__)
 
+#define MAX_CORR (SCALE_US * 100)
+
 static void hda_audio_input_timer(void *opaque) {
 
 #define B_SIZE sizeof(st->buf)
@@ -246,12 +248,16 @@ static void hda_audio_input_cb(void *opaque, int avail)
 
     int64_t to_transfer = audio_MIN(B_SIZE - (wpos - rpos), avail);
 
-//    int64_t overflow = wpos - rpos - to_transfer - (B_SIZE >> 3);
-//    if (overflow > 0) {
-//        int64_t corr = NANOSECONDS_PER_SECOND * overflow / (4 * st->as.freq);
-//        //dolog("CORR %"PRId64"\n", corr);
-//        atomic_fetch_add(&st->buft_start, corr);
-//    }
+
+    int64_t c = (wpos - rpos) + to_transfer - (B_SIZE >> 1);
+    int64_t corr = NANOSECONDS_PER_SECOND * c / (4 * st->as.freq);
+    if (corr > MAX_CORR) {
+        corr = MAX_CORR;
+    } else if (corr < -MAX_CORR) {
+        corr = -MAX_CORR;
+    }
+    //dolog("CORR %"PRId64"\n", -corr);
+    atomic_fetch_add(&st->buft_start, -corr);
 
     while (to_transfer) {
         uint32_t start = (uint32_t) (wpos & B_MASK);
@@ -334,12 +340,16 @@ static void hda_audio_output_cb(void *opaque, int avail)
 
     int64_t to_transfer = audio_MIN(wpos - rpos, avail);
 
-    int64_t overflow = wpos - rpos - to_transfer - (B_SIZE >> 3);
-    if (overflow > 0) {
-        int64_t corr = NANOSECONDS_PER_SECOND * overflow / (4 * st->as.freq);
-        //dolog("CORR %"PRId64"\n", corr);
-        atomic_fetch_add(&st->buft_start, corr);
+
+    int64_t c = (wpos - rpos) - to_transfer - (B_SIZE >> 1);
+    int64_t corr = NANOSECONDS_PER_SECOND * c / (4 * st->as.freq);
+    if (corr > MAX_CORR) {
+        corr = MAX_CORR;
+    } else if (corr < -MAX_CORR) {
+        corr = -MAX_CORR;
     }
+    //dolog("CORR %"PRId64"\n", corr);
+    atomic_fetch_add(&st->buft_start, corr);
 
     while (to_transfer) {
         uint32_t start = (uint32_t) (rpos & B_MASK);
-- 
2.15.0

