From 44efbe267e6fd7f9cb4dd1caad68c48ad9e61ac0 Mon Sep 17 00:00:00 2001
From: Martin Schrodt <martin@schrodt.org>
Date: Sun, 15 Oct 2017 15:35:38 +0200
Subject: [PATCH 19/30] ignore old fields in live migration

---
 hw/audio/hda-codec.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/hw/audio/hda-codec.c b/hw/audio/hda-codec.c
index af686eb1f3..7976c4dc93 100644
--- a/hw/audio/hda-codec.c
+++ b/hw/audio/hda-codec.c
@@ -159,6 +159,8 @@ struct HDAAudioStream {
         SWVoiceIn *in;
         SWVoiceOut *out;
     } voice;
+    uint8_t compat_buf[HDA_BUFFER_SIZE];
+    uint32_t compat_bpos;
     uint8_t buf[8192]; // size must be power of two
     int64_t rpos;
     int64_t wpos;
@@ -690,7 +692,7 @@ static void hda_audio_reset(DeviceState *dev)
 
 static const VMStateDescription vmstate_hda_audio_stream = {
     .name = "hda-audio-stream",
-    .version_id = 1,
+    .version_id = 2,
     .fields = (VMStateField[]) {
         VMSTATE_UINT32(stream, HDAAudioStream),
         VMSTATE_UINT32(channel, HDAAudioStream),
@@ -699,9 +701,8 @@ static const VMStateDescription vmstate_hda_audio_stream = {
         VMSTATE_UINT32(gain_right, HDAAudioStream),
         VMSTATE_BOOL(mute_left, HDAAudioStream),
         VMSTATE_BOOL(mute_right, HDAAudioStream),
-        VMSTATE_BUFFER(buf, HDAAudioStream),
-        VMSTATE_INT64(rpos, HDAAudioStream),
-        VMSTATE_INT64(wpos, HDAAudioStream),
+        VMSTATE_UINT32(compat_bpos, HDAAudioStream),
+        VMSTATE_BUFFER(compat_buf, HDAAudioStream),
         VMSTATE_END_OF_LIST()
     }
 };
-- 
2.15.0

