From 8628c0161cb6831bc58340f5cf21cbbb9025a1ac Mon Sep 17 00:00:00 2001
From: Martin Schrodt <martin@schrodt.org>
Date: Sun, 15 Oct 2017 18:04:13 +0200
Subject: [PATCH 26/30] checkpatch cleanup 4

Signed-off-by: Martin Schrodt <martin@schrodt.org>
---
 hw/audio/hda-codec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/audio/hda-codec.c b/hw/audio/hda-codec.c
index 38c8eb1561..ab89158bfc 100644
--- a/hw/audio/hda-codec.c
+++ b/hw/audio/hda-codec.c
@@ -162,7 +162,7 @@ struct HDAAudioStream {
     } voice;
     uint8_t compat_buf[HDA_BUFFER_SIZE];
     uint32_t compat_bpos;
-    uint8_t buf[8192]; // size must be power of two
+    uint8_t buf[8192]; /* size must be power of two */
     int64_t rpos;
     int64_t wpos;
     QEMUTimer *buft;
-- 
2.15.0

