From 018838a45eb3e0938049810121150135902c8ec1 Mon Sep 17 00:00:00 2001
From: Martin Schrodt <martin@schrodt.org>
Date: Sun, 15 Oct 2017 22:15:12 +0200
Subject: [PATCH 29/30] whitespace

Signed-off-by: Martin Schrodt <martin@schrodt.org>
---
 audio/paaudio.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/audio/paaudio.c b/audio/paaudio.c
index 9bd086de5f..e76e1e006f 100644
--- a/audio/paaudio.c
+++ b/audio/paaudio.c
@@ -345,12 +345,12 @@ static pa_stream *qpa_simple_new (
         r = pa_stream_connect_playback(stream, dev, attr,
                 PA_STREAM_INTERPOLATE_TIMING
                 | (g->conf.adjust_latency_out ? PA_STREAM_ADJUST_LATENCY : 0)
-                |PA_STREAM_AUTO_TIMING_UPDATE, NULL, NULL);
+                | PA_STREAM_AUTO_TIMING_UPDATE, NULL, NULL);
     } else {
         r = pa_stream_connect_record(stream, dev, attr,
                 PA_STREAM_INTERPOLATE_TIMING
                 | (g->conf.adjust_latency_in ? PA_STREAM_ADJUST_LATENCY : 0)
-                |PA_STREAM_AUTO_TIMING_UPDATE);
+                | PA_STREAM_AUTO_TIMING_UPDATE);
     }
 
     if (r < 0) {
@@ -414,7 +414,7 @@ static int qpa_init_out(HWVoiceOut *hw, struct audsettings *as,
     pa->ss.channels = as->nchannels;
     pa->ss.rate = as->freq;
 
-    pa->ba.tlength = tlength * pa_frame_size (&pa->ss);
+    pa->ba.tlength = tlength * pa_frame_size(&pa->ss);
     pa->ba.maxlength = -1;
     pa->ba.minreq = -1;
     pa->ba.prebuf = -1;
-- 
2.15.0

