From 27d09b98e1c9957b4f550f315672fe38858a9b0e Mon Sep 17 00:00:00 2001
From: Martin Schrodt <martin@schrodt.org>
Date: Sun, 15 Oct 2017 22:19:49 +0200
Subject: [PATCH 30/30] bigger default fraglength, to reduce PA CPU usage

Signed-off-by: Martin Schrodt <martin@schrodt.org>
---
 audio/paaudio.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/audio/paaudio.c b/audio/paaudio.c
index e76e1e006f..b46beeea92 100644
--- a/audio/paaudio.c
+++ b/audio/paaudio.c
@@ -459,7 +459,7 @@ static int qpa_init_in(HWVoiceIn *hw, struct audsettings *as, void *drv_opaque)
 
     int64_t fragsize = g->conf.fragsize;
     if (fragsize == 0) {
-        fragsize = frames_per_tick_x1000  / 2500;
+        fragsize = frames_per_tick_x1000 / 1000;
     }
     int64_t buflen = g->conf.buffer_size_in;
     if (buflen == 0) {
@@ -467,7 +467,7 @@ static int qpa_init_in(HWVoiceIn *hw, struct audsettings *as, void *drv_opaque)
     }
     int64_t maxlength = g->conf.maxlength_in;
     if (maxlength == 0) {
-        maxlength = fragsize * 4;
+        maxlength = fragsize * 2;
     }
 
     ldebug("IN internal buffer: %.2f ms (%"PRId64" frames)\n",
-- 
2.15.0

